### PR TITLE
BattMonitor: Unify common functions

### PR DESCRIPTION
**Pull request by WickedShell**
Extends #6089, unifies the read_word() interface, and then leverages it for the example of reading temperature. I included that in this PR because thats the way I'm moving to adding support for all the new/common SMBus accessors (IE Serial number, state of charge etc).

### CODE REVIEW

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus.h [hunk 1]
```diff
@@ -4,6 +4,11 @@
 #include <AP_Param/AP_Param.h>
 #include <AP_Math/AP_Math.h>
 #include "AP_BattMonitor_Backend.h"
+#include <utility>
+
+extern const AP_HAL::HAL& hal;
```
Comment (OXINARF) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus.h]: Is this needed?

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp [hunk 2]
```diff
@@ -2,6 +2,34 @@
...
 
+// read word from register
+// returns true if read was successful, false if failed
+bool AP_BattMonitor_SMBus::read_word(uint8_t reg, uint16_t& data) const
+{
+    // buffer to hold results (1 extra byte returned holding PEC)
+    const uint8_t read_size = 2 + (_pec_supported ? 1 : 0);
+    uint8_t buff[read_size];    // buffer to hold results
+
+    // read three bytes and place in last three bytes of buffer
```
Comment (OXINARF) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp]: It was already wrong, but can you correct the comment? Just having *read register* might be the better option.

File: libraries/AP_BattMonitor/AP_BattMonitor_SMBus.h (hunk 1 above)
Comment (WickedShell) [inline libraries/AP_BattMonitor/AP_BattMonitor_SMBus.h]: Hmmm I copied this in as the maxell/solo both used it (And I needed AP_HAL.h and figured there was no other reason for the weird declaration), but we appear to compile just fine with out it.... In fact it compiles if I remove it from the solo and maxell cpp files as well.... Removed.

**Comment by OXINARF** [comment]
Merged, thanks!

**Pull request merged**